app/actions/CustomActionBuilders.scala

```diff
@@ -60,6 +65,16 @@ class CustomActionBuilders(
 
   val AuthenticatedAction = PrivateAction andThen authenticated()
 
+  val NewSignInFlowAuthenticatedAction = (useNewSignIn: Option[Boolean]) =>
+    useNewSignIn.fold {
```

Comment:
also you could write this

```scala
if (useNewSignIn.contains(true)) {
  PrivateAction andThen authenticated(newSignInFlowChooseRegister)
} else {
  AuthenticatedAction
}
```

to avoid duplicating `AuthenticatedAction`, which I think makes things clearer. Or alternatively, do a `.getOrElse(false)` in `displayForm` below, and then have this function accept just a `Boolean`

Reply:
👍